src/Actions/Address/UpdateAddress.php

```diff
@@ -140,20 +140,25 @@ protected function validateData(): void
             ->first();
 
         if ($address->can_login && ($this->data['can_login'] ?? false)) {
-            if ($address->login_name
-                && array_key_exists('login_name', $this->data)
-                && ! $this->data['login_name']
+            if ($address->email
+                && array_key_exists('email', $this->data)
+                && ! $this->data['email']
             ) {
                 $errors += [
-                    'login_name' => [__('Unable to clear login name while \'can_login\' = \'true\'')],
+                    'email' => [__('Unable to clear email while \'can_login\' = \'true\'')],
                 ];
             }
-            if ($address->login_password &&
-                array_key_exists('login_password', $this->data) &&
-                ! $this->data['login_password']
+            if ($address->password &&
+                array_key_exists('password', $this->data) &&
+                ! $this->data['password']
             ) {
                 $errors += [
-                    'login_password' => [__('Unable to clear login password while \'can_login\' = \'true\'')],
+                    'password' => [__('Unable to clear password while \'can_login\' = \'true\'')],
+                ];
+            }
+            if (data_get($this->data, 'password') && ! $address->email) {
```

Comment:
just keep the required_with in ruleset, and do a $this->data['email'] ??= $address->email in prepareForValidation